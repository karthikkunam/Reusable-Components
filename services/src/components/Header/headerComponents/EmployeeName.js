import React from 'react';
import PropTypes from 'prop-types';

function EmployeeName(props) {
  const { employeeName } = props;
  return (
    <span className="navbar-nav header-info-container ml-auto">
      {employeeName && <span className="navbar-text header-name-text">Hi {employeeName}</span>}
    </span>
  );
}

EmployeeName.propTypes = {
  employeeName: PropTypes.string,
};

export default EmployeeName;
