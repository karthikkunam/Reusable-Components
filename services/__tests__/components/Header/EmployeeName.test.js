import React from 'react';
import EmployeeName from '../../../src/components/Header/headerComponents/EmployeeName';
import { shallow } from 'enzyme';

describe('EmployeeName Component', () => {
//   it('renders correctly', () => {
//     const component = shallow(<EmployeeName />);
//     expect(component).toMatchSnapshot();
//   });
  it('renders employeeName correctly', () => {
    const props = {
      employeeName: '7BOSS',
    };
    const component = shallow(<EmployeeName {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('when employeeName is undefined ', () => {
    const props = {
      employeeName: undefined,
    };
    const component = shallow(<EmployeeName {...props} />);
    expect(component).toMatchSnapshot();
  });
});
