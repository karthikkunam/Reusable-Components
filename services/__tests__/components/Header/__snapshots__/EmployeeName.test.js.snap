// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeName Component renders employeeName correctly 1`] = `
<span
  className="navbar-nav header-info-container ml-auto"
>
  <span
    className="navbar-text header-name-text"
  >
    Hi 
    7BOSS
  </span>
</span>
`;

exports[`EmployeeName Component when employeeName is undefined  1`] = `
<span
  className="navbar-nav header-info-container ml-auto"
/>
`;
